services/repository/files/commit.go

```diff
@@ -5,13 +5,65 @@ package files
 
 import (
 	"context"
+	"fmt"
 
 	asymkey_model "code.gitea.io/gitea/models/asymkey"
+	git_model "code.gitea.io/gitea/models/git"
 	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
 	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/gitrepo"
+	repo_module "code.gitea.io/gitea/modules/repository"
 	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/services/automerge"
+	"code.gitea.io/gitea/services/notify"
 )
 
+// CreateCommitStatus creates a new CommitStatus given a bunch of parameters
+// NOTE: All text-values will be trimmed from whitespaces.
+// Requires: Repo, Creator, SHA
+func CreateCommitStatus(ctx context.Context, repo *repo_model.Repository, creator *user_model.User, sha string, status *git_model.CommitStatus) error {
```

Comment:
I don't understand. I don't see any usage of this function, how it works?

Reply:
I believe it's related to #29444.

Reply:
[9c24434](https://github.com/go-gitea/gitea/pull/27151/commits/9c24434d3eefbb6130ca2c43813121d3d2c38173)